src/GovUk.Education.ExploreEducationStatistics.Admin/Services/UserManagementService.cs

```diff
@@ -338,78 +338,52 @@ public async Task<Either<ActionResult, Unit>> DeleteUser(string email)
                 {
-                    // Delete the Identity Framework user, if found.
-                    if (tuple.identityUser is not null)
-                    {
-                        await userManager.DeleteAsync(tuple.identityUser);
-                    }
+                    await userManager.DeleteAsync(identityUser);
 
-                    // Delete any invites that they may have had.
-                    usersAndRolesDbContext.UserInvites.RemoveRange(globalInvites);
-
-                    if (tuple.internalUser is not null)
-                    {
-                        await userPublicationInviteRepository.RemoveByUserEmail(email);
-                        await userReleaseInviteRepository.RemoveByUserEmail(email);
-                        await userReleaseRoleRepository.RemoveForUser(tuple.internalUser.Id);
-                        await userPublicationRoleRepository.RemoveForUser(tuple.internalUser.Id);
-                    }
+                    await userPublicationInviteRepository.RemoveByUserEmail(email);
+                    await userReleaseInviteRepository.RemoveByUserEmail(email);
 
-                    if (tuple.internalUser is { SoftDeleted: null })
-                    {
-                        tuple.internalUser.SoftDeleted = DateTime.UtcNow;
-                        tuple.internalUser.DeletedById = userService.GetUserId();
-                        tuple.internalUser.Active = false;
-                    }
+                    await userReleaseRoleRepository.RemoveForUser(activeInternalUser.Id);
+                    await userPublicationRoleRepository.RemoveForUser(activeInternalUser.Id);
 
-                    await contentDbContext.SaveChangesAsync();
-                    await usersAndRolesDbContext.SaveChangesAsync();
+                    await userRepository.SoftDeleteUser(activeInternalUser, userService.GetUserId());
                 });
             });
     }
 
-    private async Task<ApplicationUser?> GetIdentityUser(string email)
+    private async Task<Either<ActionResult, User>> GetActiveUser(string email)
     {
-        return await usersAndRolesDbContext.Users.SingleOrDefaultAsync(user => user.Email == email);
+        var activeUser = await userRepository.FindActiveUserByEmail(email);
+
+        return activeUser is null ? new NotFoundResult() : activeUser;
     }
 
-    private async Task<Either<ActionResult, Unit>> ValidateUserDoesNotExist(string email)
+    private async Task<Either<ActionResult, ApplicationUser>> GetIdentityUser(string email)
     {
-        if (await usersAndRolesDbContext.Users.AsQueryable().AnyAsync(u => u.Email!.ToLower() == email.ToLower()))
-        {
-            return ValidationActionResult(UserAlreadyExists);
-        }
+        var identityUser = await usersAndRolesDbContext.Users.SingleOrDefaultAsync(user => user.Email == email);
+
+        return identityUser is null ? new NotFoundResult() : identityUser;
+    }
```

Comment:
This could use `SingleOrNotFoundAsync`, replacing it with:

```csharp
private async Task<Either<ActionResult, ApplicationUser>> GetIdentityUser(string email) => await usersAndRolesDbContext.Users.SingleOrNotFoundAsync(user => user.Email == email);
```